### PR TITLE
Multilingual BART -

### PR DESCRIPTION
**Pull request by sshleifer**
This adds `mbart-en-ro` model, a BART variant finetuned on english-romanian translation.
### TODO
- [x] (docs) pretrained_model.rst
- [ ] (docs) README.md
- [ ] (docs) bart.rst

### Differences with Bart
`config.normalize_before`: all the `LayerNorm` calls happen before attention calls
`config.add_final_layer_norm`: There is one extra layer_norm in the decoder
`config.scale_embedding`: embeddings are multiplied by 32 (`sqrt(d_model=1024)`)

### Future PRs
- The model returns the same variables as fairseq, but the tokenizer is not yet at parity with fairseq. This is the next PR in the pipeline.
- the `mbart-large-cc25` (no finetuning) model has a very different state dict. Also WIP.

### Misc
- the link_tester got angry about me not typing out URLs in this PR. Unclear why it didn't happen earlier.
Needs documentation but unclear where to put it.

### CODE REVIEW

File: src/transformers/configuration_bart.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "bart-large-mnli": "https://s3.amazonaws.com/models.huggingface.co/bert/facebook/bart-large-mnli/config.json",
     "bart-large-cnn": "https://s3.amazonaws.com/models.huggingface.co/bert/facebook/bart-large-cnn/config.json",
     "bart-large-xsum": "https://s3.amazonaws.com/models.huggingface.co/bert/facebook/bart-large-xsum/config.json",
+    "mbart-large-en-ro": "https://s3.amazonaws.com/models.huggingface.co/bert/facebook/mbart-large-en-ro/config.json",
```
Comment (julien-c) [inline src/transformers/configuration_bart.py:30]: should probably be `facebook/mbart-large-en-ro`

**Review by julien-c — APPROVED** [review]

File: src/transformers/modeling_bart.py [hunk 2]
```diff
@@ -360,16 +382,20 @@ def forward(
         )
         x = F.dropout(x, p=self.dropout, training=self.training)
         x = residual + x
+        if not self.normalize_before:
```
Comment (patrickvonplaten) [inline src/transformers/modeling_bart.py:385]: Not a big fan of the `self.normalize_before `and after statements. Seems hard to avoid though. Maybe Bart could use some `BartAttention` and `BartIntermediate` Layer classes similar to Bert's implementation and then a much smaller `BartEncoder` and `BartMultiLingualEncoder` Layer which uses the `BartAttention` and `BartIntermediate` layers and puts the LayerNorm differently in the correct place. But I guess that would also require to change the way the weights are loaded into the model, so not ideal either ...

Comment (sshleifer) [reply src/transformers/modeling_bart.py:385]: Yeah I spent an hour trying to move it around. 

Also tried to dynamically define `nn.ModuleList` with layer_norm in the correct place.

This felt more readable.

File: src/transformers/modeling_bart.py [hunk 3]
```diff
@@ -478,6 +506,9 @@ def forward(
 
             if self.output_past:
                 next_decoder_cache.append(layer_past.copy())
+
+            if self.layer_norm and (idx == len(self.layers) - 1):  # last layer of mbart
+                x = self.layer_norm(x)
```
Comment (thomwolf) [inline src/transformers/modeling_bart.py:511]: I think there is something to fix on the following line (missed it on the BART PR).

The `all_hidden_states` list outputted by our models comprises `n_layer+1` elements with the first being the hidden-state after the word embedding (will be 2 times this for encoder-decoder).

I think the output of the word embeddings is missing here.

We are supposed to have a test on this so you should check the tests is activated for the encoder-decoder models (and activate/tweak it to work it if it's not).

**Review by thomwolf — COMMENTED** [review]
Ok!

Just check the hidden-states output comment and it's good to go.

File: src/transformers/configuration_bart.py (hunk 1 above)
Comment (sshleifer) [reply src/transformers/configuration_bart.py:30]: I dont see the difference

File: src/transformers/modeling_bart.py (hunk 3 above)
Comment (sshleifer) [reply src/transformers/modeling_bart.py:511]: Good catch! I fixed `all_hidden_states` here + made an issue to fix the test, which I will address in a separate PR.

**Comment by h324yang** [comment]
Hi, is there any reason that cc-25 was removed and only the fine-tuned one is kept? any way I can quickly enable that? thanks

**Comment by sshleifer** [comment]
When the authors released the CC25 checkpoint, it was shaped differently than `mbart-large-en-ro` and I am not clear on whether that is fixed yet.
See https://github.com/pytorch/fairseq/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+mbart
